### DISCUSSION TITLE
setValue seems not to work correctly with array's

### DISCUSSION THREAD

**Discussion by phjdreef** [react-hook-form/react-hook-form#4884] [Q&A]
I want to use the following method to update the values in an array, but after the first "set", the value isn't changing anymore:
Based on a sample provided by you, with the latest version of react-form-hook:

A total reset of the form is working, but a little annoying while typing.

https://codesandbox.io/s/usefieldarray-with-setvalue-forked-8iq9b
The third "setValue"  is not setting the correct values.

BTW, I love your package!!

grtz,

Paul

**Comment by bluebill1049** [comment]
hey Paul, what's not working on the `setValue`?

**Reply by phjdreef** [reply]
In the example, you see that the values are not resetted.
The values provided in the second and third "update" are not showing coirrectly:

`{
          value: 0,
          nestedArray: [
            {
              value: 8
            }
          ]
        },
        {
          value: 9,
          nestedArray: [
            {
              value: 8
            }
          ]
        },
        {
          value: 20,
          nestedArray: [
            {
              value: 78
            }
          ]
        }`

You see:
![image](https://user-images.githubusercontent.com/61777220/115505187-5a9e5300-a279-11eb-8ba8-ae326dfdd75e.png)

**Reply by bluebill1049** [reply]
thanks, this PR should fix it: https://github.com/react-hook-form/react-hook-form/pull/4861

**Reply by phjdreef** [reply]
Ok, seems ok with me. I will try out the next version, ok ?

**Reply by bluebill1049** [reply]
sounds good, i will leave you a beta version once the next patch goes out.

**Reply by bluebill1049** [reply]
you want to try this version: `v7.2.3-beta.0`?
